src/crm/activecampaign.ts

```diff
@@ -0,0 +1,300 @@
+
+
+    } catch (err) {
+      console.error("Error fetching contact:", err);
+      return null;
+    }
+  };
+
+  createContact = async (bodyContent: string): Promise<string> => {
+    const res = await fetch(`${url}/api/3/contacts`, {
+      method: "POST",
+      headers: this.headers,
+      body: bodyContent,
+    });
+    if (!res.ok) throw new Error(`Failed to create contact: ${res.statusText}`);
+    const data = await res.json();
+    return data.contact.id;
+  };
+
+  updateContact = async (contactid: string, bodyContent: string): Promise<string> => {
+    const res = await fetch(`${url}/api/3/contacts/${contactid}`, {
+      method: "PUT",
+      headers: this.headers,
+      body: bodyContent,
+    });
+
+    if (!res.ok) throw new Error(`Failed to update contact: ${res.statusText}`);
+    const data = await res.json();
+    return data.contact.id;
+  };
+
+  subscribeToList = async (contactid: string, listid: string): Promise<void> => {
+    const res = await fetch(`${url}/api/3/contactLists`, {
+      method: "POST",
+      headers: this.headers,
+      body: JSON.stringify({
+        contactList: {
+          list: listid,
+          contact: contactid,
+          status: 1
+        },
+      }),
+    });
+
+    if (!res.ok) throw new Error(`Failed to subscribe to list: ${res.statusText}`);
+    console.log(`Subscribed contact ${contactid} to list ${listid}`);
+  };
+
+  //The tag must already exist, default?
+  addTagsToContact = async (contactId: string, tagIds: string): Promise<void> => {
+    const ids = tagIds.replace(/\s+/g, "").split(",");
+    for (const tagId of ids) {
+      const res = await fetch(`${url}/api/3/contactTags`, {
+        method: "POST",
+        headers: this.headers,
+        body: JSON.stringify({
+          contactTag: {
+            contact: contactId,
+            tag: tagId,
+          },
+        }),
+      });
+
+    if (!res.ok) {
+      const errorData = await res.json();
+      throw new Error(`Failed to add tag ${tagId}: ${JSON.stringify(errorData)}`);
+    }
+    console.log(`Tag ${tagId} added to contact ${contactId}`);
+  }
+};
+
+  handleActionContact = async (
+    message: ActionMessage
+  ): Promise<handleResult | boolean> => {
+    const email = message.contact.email;
+
+    console.log("Processing action:", message.action.id, "testing:", message.action.testing);
+
+      if (this.verbose) {
+      console.log(JSON.stringify(message, null, 2));
+    }
+    try {
+
+      let camp = await this.fetchCampaign(message.campaign.id);
+      const { listid,
+        tagids,
+        action_id_field,
+        ref_field,
+        data_source,
+        zip_field } = camp.config?.component?.sync || {};
+
+      if (!tagids || !action_id_field || !ref_field) {
+        console.error("Missing required configuration for ActiveCampaign sync");
+        return false;
+      };
+
+      let contactid = await this.fetchContact(email);
+
+      const contactPayload: ContactPayload = {
```

Comment:
I don't get why the payload is different if id or email, can you add a comment for future ourselves?